server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -491,9 +497,22 @@ private void recoverFromRemoteStore(IndexShard indexShard) throws IndexShardReco
                 }
             }
 
-            // This creates empty trans-log for now
-            // ToDo: Add code to restore from remote trans-log
-            bootstrap(indexShard, store);
+            if (repository != null) {
+                FileTransferTracker fileTransferTracker = new FileTransferTracker(shardId);
+                assert repository instanceof BlobStoreRepository : "repository should be instance of BlobStoreRepository";
+                BlobStoreRepository blobStoreRepository = (BlobStoreRepository) repository;
+                TranslogTransferManager translogTransferManager = new TranslogTransferManager(
+                    new BlobStoreTransferService(
+                        blobStoreRepository.blobStore(),
+                        indexShard.getThreadPool().executor(ThreadPool.Names.TRANSLOG_TRANSFER)
+                    ),
+                    blobStoreRepository.basePath().add(shardId.getIndex().getUUID()).add(String.valueOf(shardId.id())),
+                    fileTransferTracker,
+                    fileTransferTracker::exclusionFilter
+                );
+                RemoteFsTranslog.download(translogTransferManager, indexShard.shardPath().resolveTranslog());
+            }
+
```

Comment:
shall we move this to a method `syncTranslogFromRemoteTranslogStore` for ease of read?